web_domain_field/README.rst

```diff
@@ -0,0 +1,133 @@
+================
+Web Domain Field
+================
+
+.. !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+   !! This file is generated by oca-gen-addon-readme !!
+   !! changes will be overwritten.                   !!
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+.. |badge1| image:: https://img.shields.io/badge/maturity-Beta-yellow.png
+    :target: https://odoo-community.org/page/development-status
+    :alt: Beta
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fweb-lightgray.png?logo=github
+    :target: https://github.com/OCA/web/tree/12.0/web_domain_field
+    :alt: OCA/web
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/web-12-0/web-12-0-web_domain_field
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/162/12.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+In order to mitigate limitations with domains in views, now you can define
+a computed field in a domain attribute and It's related method will return
+the domain to apply.
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+When you define a view you can specify on the relational fields a domain
+attribute. This attribute is evaluated as filter to apply when displaying
+existing records for selection.
+
+.. code-block:: xml
+
+   <field name="product_id" domain="[('type','=','product')]"/>
+
+The value provided for the domain attribute must be a string representing a
+valid Odoo domain. This string is evaluated on the client side in a
+restricted context where we can reference as right operand the values of
+fields present into the form and a limited set of functions.
+
+In this context it's hard to build complex domain and we are facing to some
+limitations as:
+
+ * The syntax to include in your domain a criteria involving values from a
+   x2many field is complex.
+ * The right side of domain in case of x2many can involve huge amount of ids
+   (performance problem).
+ * Domains computed by an onchange on an other field are not recomputed when
+   you modify the form and don't modify the field triggering the onchange.
+ * It's not possible to extend an existing domain. You must completely redefine
+   the domain in your specialized addon
+ * etc...
+
```

Comment:
but the same case can be achieved by specifying domain in field definition. For the example you have given, it can be achieved by 
```
testfield = fields.Char()
product_id = fields.Many2one('product.product',domain="[('type', '=', 'product'), ('name', 'like', testfield)]")
```
Maybe I am missing the whole picture here.

Reply:
@omalbastin The point is that the domain field can change at runtime.